press/press/doctype/press_role_permission/press_role_permission.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2024, Frappe and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+
+from press.api.client import dashboard_whitelist
+
+
+class PressRolePermission(Document):
+	# begin: auto-generated types
+	# This code is auto-generated. Do not modify anything in this block.
+
+	from typing import TYPE_CHECKING
+
+	if TYPE_CHECKING:
+		from frappe.types import DF
+
+		release_group: DF.Link | None
+		role: DF.Link
+		server: DF.Link | None
+		site: DF.Link | None
+		team: DF.Link
+	# end: auto-generated types
+
+	dashboard_fields = ["site", "release_group", "server", "role"]
+
+	def before_insert(self):
+		if frappe.db.exists(
+			"Press Role Permission",
+			{
+				"role": self.role,
+				"team": self.team,
+				"site": self.site,
+				"release_group": self.release_group,
+				"server": self.server,
+			},
+		):
+			frappe.throw("Role Permission already exists")
+
+	@dashboard_whitelist()
+	def delete(self):
+		super().delete()
```

Comment:
should we have a check for who can delete/add role permisions?

Reply:
Simple approach would be to only allow the team owner. But if we should let others have such admin like privileges maybe we can create special roles with such privileges.